src/devices.js

```diff
@@ -239,7 +239,46 @@ function installStep(step) {
         global.mainEvent.emit(
           "user:action",
           global.installConfig.user_actions[step.action],
-          resolve
+          () => {
+            switch (step.action) {
+              case "recovery":
+              case "system":
+                global.mainEvent.emit("user:write:working", "particles");
+                global.mainEvent.emit(
+                  "user:write:status",
+                  "Waiting for device",
+                  true
+                );
+                global.mainEvent.emit(
+                  "user:write:under",
+                  "Adb is scanning for devices"
+                );
+                adb
+                  .waitForDevice()
+                  .then(resolve)
```

Comment:
Since this is wrapped in a promises, you can just return adb.waitForDevice() no need for .then